lms/djangoapps/grades/course_grade_factory.py

```diff
@@ -193,26 +197,27 @@ def _update(user, course_data, force_update_subsections=False):
                 passed=course_grade.passed,
             )
 
-        COURSE_GRADE_CHANGED.send_robust(
-            sender=None,
-            user=user,
-            course_grade=course_grade,
-            course_key=course_data.course_key,
-            deadline=course_data.course.end,
-        )
-        if course_grade.passed:
-            COURSE_GRADE_NOW_PASSED.send(
-                sender=CourseGradeFactory,
-                user=user,
-                course_id=course_data.course_key,
-            )
-        else:
-            COURSE_GRADE_NOW_FAILED.send(
-                sender=CourseGradeFactory,
+        if send_grade_signals:
```

Comment:
why not use `should_persist` as the flag to send the signal? or is there another use case where we would want to send the signal?